src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -352,16 +349,27 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
             }
         }
 
-        // assumes all same case.        
+       
         private bool ArgumentMatch(string arg, string formal)
         {
-            if (arg[0] != '/' && arg[0] != '-')
+            // Full name format, eg: --assembly
+            if (arg.Length <= 1 || arg[0] != '-' || arg[1] != '-' )
```

Comment:
I think this function requires `arg.Length < 4`.

Reply:
@zhenlan If argument is "--" or "-", all ArgumentMatch will return false and there will be a warning message from line 140: "Warning: Ignoring invalid command line argument: '--'. " So do we still need to check arg.Length<3

Reply:
Not sure what you mean by line 140 as I don't see anything meaningful.

We should have the function self-contained so we don't worry if something is checked somewhere. An external check can be removed somehow for another reason. So, yes, please do check arg.Length here.